controllers/azurejson_machinepool_controller.go

```diff
@@ -194,3 +194,7 @@ func (r *AzureJSONMachinePoolReconciler) Reconcile(ctx context.Context, req ctrl
 
 	return ctrl.Result{}, nil
 }
+
+var getClient = func(auth azure.Authorizer) (identities.Client, error) {
```

Comment:
This "patch" allows for this to be overridden during testing but not in production.